torch/testing/_internal/common_distributed.py

```diff
@@ -359,9 +360,32 @@ def _spawn_processes(self):
         proc = torch.multiprocessing.get_context("spawn").Process
         self._start_processes(proc)
 
+    class Event(Enum):
+        GET_TRACEBACK = 1
+
+    @staticmethod
+    def _event_listener(pipe):
+        while True:
+            event = pipe.recv()
+            if event == MultiProcessTestCase.Event.GET_TRACEBACK:
+                # Return traceback to the parent process.
+                with tempfile.NamedTemporaryFile(mode='r+') as tmp_file:
+                    faulthandler.dump_traceback(tmp_file)
+                    # Flush buffers and seek to read from the beginning
+                    tmp_file.flush()
+                    tmp_file.seek(0)
+                    pipe.send(tmp_file.read())
+
     @classmethod
     def _run(cls, rank, test_name, file_name, pipe, faulthandler_file_name):
         self = cls(test_name)
+
+        # Start event listener thread.
+        threading.Thread(
+            target=MultiProcessTestCase._event_listener,
+            args=(pipe,),
+            daemon=True).start()
```

Comment:
How is this thread destroyed at the end? Does it need to be?

Reply:
I used `daemon=True` so we don't have to worry about that.